File: java/src/test/java/io/binac/leetcode/RemoveDuplicatesFromSortedArrayTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;

import static org.junit.jupiter.api.Assertions.*;

class RemoveDuplicatesFromSortedArrayTest {
    private static final RemoveDuplicatesFromSortedArray.Solution1 SOLUTION1 = new RemoveDuplicatesFromSortedArray.Solution1();

    private final int[] NUMS1 = {1, 1, 2};
    private final int[] EXPECTED1 = {1, 2};

    private final int[] NUMS2 = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
    private final int[] EXPECTED2 = {0, 1, 2, 3, 4};

    @Test
    void test1() {
        int size1 = SOLUTION1.removeDuplicates(NUMS1);
        assertEquals(EXPECTED1.length, size1);
        assertArrayEquals(EXPECTED1, Arrays.copyOf(NUMS1, size1));

        int size2 = SOLUTION1.removeDuplicates(NUMS2);
        assertEquals(EXPECTED2.length, size2);
        assertArrayEquals(EXPECTED2, Arrays.copyOf(NUMS2, size2));
    }
}

File: cpp/unittests/RemoveDuplicatesFromSortedArrayTest.cpp
#include "RemoveDuplicatesFromSortedArray.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(RemoveDuplicatesFromSortedArray, Solution26_1) {
  std::vector<int> Nums1{1, 1, 2};
  const int Size1 = 2;
  ASSERT_EQ(Size1, Solution26_1().removeDuplicates(Nums1));
  const std::vector<int> Expected1{1, 2};
  EXPECT_EQ(Expected1, std::vector<int>(Nums1.begin(), Nums1.begin() + Size1));

  std::vector<int> Nums2{0, 0, 1, 1, 1, 2, 2, 3, 3, 4};
  const int Size2 = 5;
  ASSERT_EQ(Size2, Solution26_1().removeDuplicates(Nums2));
  const std::vector<int> Expected2{0, 1, 2, 3, 4};
  EXPECT_EQ(Expected2, std::vector<int>(Nums2.begin(), Nums2.begin() + Size2));
}